lib/matplotlib/ticker.py

```diff
@@ -2933,6 +2934,12 @@ def __call__(self):
             return []
 
         if self.ndivs is None:
+            if self.axis.__name__ == 'xaxis':
+                self.ndivs = mpl.rcParams['xtick.minor.ndivs']
+            else:
+                self.ndivs = mpl.rcParams['ytick.minor.ndivs']
```

Comment:
This won't work for 3D; it normally uses the *x* setting for *z*.
Also, better to use `self.axis.axis_name` instead.